site/docs/master/faq.md

```diff
@@ -0,0 +1,216 @@
+To read the logs for a test failure, you can fnd the log file within the results tarball from Sonobuoy (`plugins/e2e/results/global/e2e.log`) or you can use the `results` command to show details of test failures.
+For example, the following commands retrieve the results tarball and then use [jq][jq] to return an object for each test failure with the failure message and the associated stdout.
+
+```
+outfile=$(sonobuoy retrieve) && \
+  sonobuoy results --mode detailed --plugin e2e $outfile |  jq '.  | select(.status == "failed") | .details'
+```
+
+Carefully read the test logs to see if anything stands out which could be the cause of the failure.
+For example: Were there difficulties when contacting a particular service? Are there any commonalities in the failed tests due to a particular feature? 
+Often, the test logs will provide enough detail to allow you to determine why a test failed.
+
+If you need more information, Sonobuoy also queries the cluster upon completion of plugins.
+The details collected allow you to see the state of the cluster and whether there were any issues.
+For example: Did any of the nodes have memory pressure? Did the scheduler pod go down?
+
+As a final resort, you can also read the upstream test code to determine what actions were being performed at the point when the test failed.
+If you decide to take this approach, you must ensure that you are reading the version of the test code that corresponds to your test image.
+You can verify which version of the test image was used by inspecting the plugin definition which is available in the results tarball in `plugins/e2e/definition.json` under the key `Definition.spec.image`.
+For example, if the test image was `gcr.io/google-containers/conformance:v1.15.3`, you should read the code at the corresponding [v1.15.3 tag in GitHub][kubernetes-1.15.3].
+All the tests can be found within the `test/e2e` directory in the Kubernetes repository.
+
+### How can I run the E2E tests with certain test framework options set? What are the available options?
+How you provide options to the E2E test framework and determining which options you can set depends on which version of Kubernetes you are testing.
+
+To view the available options that you can set when running the tests, you can run the test executable for the conformance image you will be using as follows:
+
+```
+KUBE_VERSION=<Kubernetes version you are using>
+docker run -it gcr.io/google-containers/conformance:$KUBE_VERSION ./e2e.test --help
+```
+
+If you are running Kubernetes v1.16.0 or greater, a new feature was included in this release which makes it easier to specify your own options.
+This new feature allows arbitrary options to be specified when the tests are invoked.
+To use this, you must enable the new feature with the flag `--plugin-env=e2e.E2E_USE_GO_RUNNER=true` and then you can provide your options with the flag `--plugin-env=e2e.E2E_EXTRA_ARGS`.
```

Comment:
`E2E_USE_GO_RUNNER` is the default now; we should make that clear instead of making it seem like that has to be manually set.

Maybe something like
```
... you must ensure the env var `E2E_USE_GO_RUNNER=true` is set (this is the default from the CLI and manually setting this only required if working from a saved YAML file...
```

Feel free to tweak the wording.